packages/plugin-core/src/commands/pods/NotionExportPodCommand.ts

```diff
@@ -0,0 +1,259 @@
+import { ErrorFactory, NoteProps, ResponseUtil } from "@dendronhq/common-all";
+import {
+  Client,
+  ConfigFileUtils,
+  createRunnableNotionV2PodConfigSchema,
+  ExportPodV2,
+  ExternalConnectionManager,
+  ExternalService,
+  isRunnableNotionV2PodConfig,
+  JSONSchemaType,
+  NotionConnection,
+  NotionExportPodV2,
+  NotionExportReturnType,
+  NotionFields,
+  NotionV2PodConfig,
+  Page,
+  PodV2Types,
+  RunnableNotionV2PodConfig,
+  TitlePropertyValue,
+} from "@dendronhq/pods-core";
+import _ from "lodash";
+import path from "path";
+import * as vscode from "vscode";
+import { ProgressLocation, window } from "vscode";
+import { QuickPickHierarchySelector } from "../../components/lookup/HierarchySelector";
+import { PodUIControls } from "../../components/pods/PodControls";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { VSCodeUtils } from "../../vsCodeUtils";
+import { getExtension } from "../../workspace";
+import { BaseExportPodCommand } from "./BaseExportPodCommand";
+
+/**
+ * VSCode command for running the Notion Export Pod. It is not meant to be
+ * directly invoked throught the command palette, but is invoked by
+ * {@link ExportPodV2Command}
+ */
+export class NotionExportPodCommand extends BaseExportPodCommand<
+  RunnableNotionV2PodConfig,
+  NotionExportReturnType
+> {
+  public key = "dendron.notionexport";
+
+  public constructor() {
+    super(new QuickPickHierarchySelector());
+  }
+
+  public createPod(
+    config: RunnableNotionV2PodConfig
+  ): ExportPodV2<NotionExportReturnType> {
+    return new NotionExportPodV2({
+      podConfig: config,
+    });
+  }
+
+  public getRunnableSchema(): JSONSchemaType<RunnableNotionV2PodConfig> {
+    return createRunnableNotionV2PodConfigSchema();
+  }
+
+  async gatherInputs(
+    opts?: Partial<NotionV2PodConfig & NotionConnection>
+  ): Promise<RunnableNotionV2PodConfig | undefined> {
+    let apiKey: string | undefined = opts?.apiKey;
+    let connectionId: string | undefined = opts?.connectionId;
+
+    // Get an Notion API Key
+    if (!apiKey) {
+      const mngr = new ExternalConnectionManager(getExtension().podsDir);
+      // If the apiKey doesn't exist, see if we can first extract it from the connectedServiceId:
+      if (opts?.connectionId) {
+        const config = mngr.getConfigById<NotionConnection>({
+          id: opts.connectionId,
+        });
+
+        if (!config) {
+          window.showErrorMessage(
+            `Couldn't find service config with the passed in connection ID ${opts.connectionId}.`
```

Comment:
Maybe let's enhance this msg a bit to make it more user friendly:

"Couldn't find service config with the specified connection ID ${id}. Ensure that the connection ID is correct. You can check existing connections with the `Dendron: Configure Service Connection` command.